test/unit/lib/aliases.js

```diff
@@ -0,0 +1,354 @@
+							return configuredAliases[featureName];
+						});
+						return resolver({
+							resolved_name_a: {
+								flags: ["always"]
+							},
+							alias_name_b: {
+								flags: ["gated"]
+							}
+						}).then(function(resolved) {
+							assert.deepEqual(setsToArrays(resolved), {
+								resolved_name_a: {
+									flags: ["always"]
+								},
+								resolved_name_c: {
+									flags: ["gated"],
+									aliasOf: ["alias_name_b"]
+								},
+								resolved_name_b: {
+									flags: ["gated"],
+									aliasOf: ["alias_name_b"]
+								}
+							});
+						});
+					});
+
+					it("should handle cases where a resolver function cannot resolve a name so returns undefined by passing the polyfill identifier to the next function", () => {
+						const configuredAliases = {
+							"alias_name_a": ["resolved_name_a", "resolved_name_b"],
+							"alias_name_b": ["resolved_name_c", "resolved_name_b"]
+						};
+
+						const resolver = aliasResolver(
+
+							// Map only first_alias_name_a to another alias
+							function(name) {
+								return (name === "first_alias_name_a") ? ["alias_name_a"] : undefined;
+							},
+							function(name) {
+								return configuredAliases[name];
+							}
+						);
+
+						return resolver({
+							first_alias_name_a: {
+								flags: ["always"]
+							},
+							alias_name_b: {
+								flags: ["gated"]
+							}
+						}).then(function(resolved) {
+							assert.deepEqual(setsToArrays(resolved), {
+								resolved_name_a: {
+									flags: ["always"],
+									aliasOf: ["alias_name_a", "first_alias_name_a"]
+								},
+								resolved_name_b: {
+									flags: ["always", "gated"],
+									aliasOf: ["alias_name_a", "alias_name_b", "first_alias_name_a"]
+								},
+								resolved_name_c: {
+									flags: ["gated"],
+									aliasOf: ["alias_name_b"]
+								}
+							});
+						});
+					});
+
+					it('should be able to resolve an alias to one of the other inputs', () => {
+						const resolver = aliasResolver(
+							function(name) {
+								return (name === 'alias_name_a') ? ['name_b'] : undefined;
+							}
+						);
+
+						return resolver({
+							alias_name_a: {
+								flags: new Set(["always"])
+							},
+							name_b: {
+								flags: new Set(["gated"])
+							}
+						}).then(function(resolved) {
+							assert.deepEqual(setsToArrays(resolved), {
+								name_b: {
+									flags: ["always", "gated"],
+									aliasOf: ['alias_name_a']
+								}
+							});
+						});
+					});
+
+					it('should be able to resolve an alias to itself', () => {
+						const resolver = aliasResolver(
+							function(name) {
+								return [name];
+							}
+						);
+
+						return resolver({
+							name_a: {},
+							name_b: {}
+						}).then(function(resolved) {
+							assert.deepEqual(setsToArrays(resolved), {
+								name_a: {},
+								name_b: {}
+							});
+						});
+					});
+				});
+			});
+
+
+		});
+
+	});
+
+});
```

Comment:
There's a bit of extraneous whitespace in these closing braces.  Also, personally I preferred the more compressed format for the data which make it more likely that you could fit the entire test on the screen at once, but it's not a big point, and I don't really care that much